engine/src/main/java/org/terasology/rendering/opengl/fbms/DynamicFBM.java

```diff
@@ -0,0 +1,163 @@
+            disposeAllFBOs();
+            createFBOs();
+        }
+    }
+
+    private void disposeAllFBOs() {
+        FBO fbo;
+        for (ResourceUrn urn : fboConfigs.keySet()) {
+            fbo = fboLookup.get(urn);
+            if (fbo != null) {
+                fbo.dispose();
+                fboLookup.remove(urn);
+
+            } else {
+                logger.error("Failed to delete FBO '" + urn + "': it doesn't exist!");
+            }
+        }
+    }
+
+    private void createFBOs() {
+        for (FBOConfig fboConfig : fboConfigs.values()) {
+            generate(fboConfig, fullScale.multiplyBy(fboConfig.getScale()));
+        }
+        notifySubscribers();
+    }
+
+    /**
+     * Returns the content of the color buffer of the FBO "sceneFinal", from GPU memory as a ByteBuffer.
+     * If the FBO "sceneFinal" is unavailable, returns null.
+     *
+     * @return a ByteBuffer or null
+     */
+    public ByteBuffer getSceneFinalRawData() {
+        FBO fboSceneFinal = getFBO(DefaultDynamicFBOs.ReadOnlyGBuffer.getResourceUrn());
+        if (fboSceneFinal == null) {
+            logger.error("FBO sceneFinal is unavailable: cannot return data from it.");
+            return null;
+        }
+
+        ByteBuffer buffer = BufferUtils.createByteBuffer(fboSceneFinal.width() * fboSceneFinal.height() * 4);
+
+        fboSceneFinal.bindTexture();
+        GL11.glGetTexImage(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA, GL11.GL_UNSIGNED_BYTE, buffer);
+        FBO.unbindTexture();
+
+        return buffer;
+    }
+
+    private void updateFullScale() {
+        if (postProcessor.isNotTakingScreenshot()) {
+            fullScale = new FBO.Dimensions(Display.getWidth(), Display.getHeight());
+            if (renderingConfig.isOculusVrSupport()) {
+                fullScale.multiplySelfBy(OculusVrHelper.getScaleFactor());
+            }
+        } else {
+            fullScale = new FBO.Dimensions(
+                    renderingConfig.getScreenshotSize().getWidth(Display.getWidth()),
+                    renderingConfig.getScreenshotSize().getHeight(Display.getHeight())
+            );
+        }
+
+        fullScale.multiplySelfBy(renderingConfig.getFboScale() / 100f);
+    }
+
+    /**
+     * Sets an internal reference to the PostProcessor instance. This reference is used to inform the PostProcessor
+     * instance that changes have occurred and that it should refresh its references to the FBOs it uses.
+     *
+     * @param postProcessor a PostProcessor instance
+     */
+    public void setPostProcessor(PostProcessor postProcessor) {
+        this.postProcessor = postProcessor;
+    }
+
+    @Override
+    public void allocateFBO(FBOConfig fboConfig) {
+        if (fboUsageCountMap.containsKey(fboConfig.getResourceUrn())) {
+            throw new IllegalArgumentException("There is already an FBO inside DynamicFBM, named as: " + fboConfig.getResourceUrn());
+        }
```

Comment:
I wouldn't throw an exception in all cases here. If the FBOConfig of the existing FBO and the one provided in input are perfectly equivalent the reference count should increase but no new FBO should be generated. The exception should be thrown only if the two FBOConfigs are not equivalent but use the same fboName.